### PR TITLE
Rebase to kube 1.18

### PR DESCRIPTION
**Pull request by damemi**

### CODE REVIEW

**Comment by damemi** [comment]
/hold
this is currently a POC based on the prebase branches that have merged in library-go, api, and client-go

File: pkg/operator/configmetrics/configmetrics.go [hunk 1]
```diff
@@ -48,3 +48,7 @@ func (m *configMetrics) Collect(ch chan<- prometheus.Metric) {
 		ch <- g
 	}
 }
+
+func (m *configMetrics) FQName() string {
+	return "cluster_master_schedulable"
```
Comment (soltysh) [inline pkg/operator/configmetrics/configmetrics.go]: Maybe worth making this a constant rather than repeating it in two places?

**Comment by damemi** [comment]
Currently pushed with the changes from https://github.com/openshift/library-go/pull/748 in my fork of library-go, need that to get in to the prebase branch

**Comment by damemi** [comment]
Reverted https://github.com/openshift/cluster-kube-scheduler-operator/pull/226/commits/630f2e5c218e9fc842d493b96fcc0918b93e4d96 to see if that's breaking tests

**Comment by soltysh** [comment]
> Reverted [630f2e5](https://github.com/openshift/cluster-kube-scheduler-operator/commit/630f2e5c218e9fc842d493b96fcc0918b93e4d96) to see if that's breaking tests

That'd be surprising but let's see :man_shrugging:

**Comment by damemi** [comment]
/retest
I think that switch to k8smetrics was breaking it. The cluster was failing with some weird logs around prometheus that pointed to the area where we register that metric.

**Comment by damemi** [comment]
/hold cancel
I think using the old metrics is still fine, but would like to switch at some point so we can inherit those methods

**Pull request merged**